libs/pricing_model/src/lib.rs

```diff
@@ -0,0 +1,338 @@
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+/// Maintains latest [`Segment`] for each of the branches, faciliates the pricing model
+/// calculations.
+pub struct Storage<K: 'static> {
+    segments: Vec<Segment>,
+    branches: HashMap<K, usize>,
+}
+
+/// State of a branch.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Segment {
```

Comment:
It's a bit confusing to store the logical size as part of the segment, if a segment represents a range of LSNs. I was confused by that myself even though I wrote this, but in hindsight, perhaps it would be more clear to have Points and Segments, so that a Point represents a particular LSN on a timeline, and and a Segment represents a range of LSNs, between two Points.

(If you got this working as it is, no need to start changing it now, but maybe in the future..)

Reply:
(after reading the PR further): I see you actually did something like that with the `updates` and `Command` stuff. Makes sense! You could move that here, I think.

Reply:
I am quite fuzzy on how well it works as discussed. Testing this in integration tests is quite the chore, as it's LSN dependent and those might be unstable so there cannot really be any direct comparisons. Well, should try that still with CI. Failed to get unstable results locally, but it's likely I failed to apply the correct TZ and locale settings.

Reply:
Ended up with bunch of comparisons in the tests without any actual data, since CI and my local results differ.